docs_src/design/adr/0027-URIs for Files.md

```diff
@@ -0,0 +1,113 @@
+
+## Proposed Design
+This ADR proposes a new helper function for loading files be added to `go-mod-bootstrap`. This function will provide the logic for loading a file either from local file system (as is today) or from a remote location. As stated in the UCR, only **HTTP** and **HTTPS** URIs will be supported. For **HTTPS**, certificate validation will be performed using the system's built-in trust anchors. The docker images for all services will have the CA certs installed as is done here in App Service Configurable's [Dockerfile](https://github.com/edgexfoundry/app-service-configurable/blob/v2.3.0/Dockerfile#L46).
+
+### Authentication 
+
+#### basic-auth in URI
+
+While not recommended, users will be able to specify **basic-auth** (`<username>:<password>@`) in the URI in plain text. While this is ok network wise when using HTTPS, it isn't good practice to have these credentials specified in configuration or other service files where the URI is specified.
+
+!!! example - "Example plain text `basic-auth` in URI located in configuration"
+    ```toml
+    [UoM]
+    UoMFile = "https://myuser:mypassword@example.com/uom.yaml"
+    ```
+
+#### Secure Credentials
+
+In order to provide a secure way for users to specify credentials, the `edgexSecretName` query parameter can be specified on the URI. This parameter specifies a Secret Name from the service's Secret Store where the credentials reside and will be processed by the new helper function.
+
+!!! example - "Example URI with `edgexSecretName` query parameter"
+    ```toml
+    [UoM]
+    UoMFile = "https://example.com/uom.yaml?edgexSecretName=mySecretName"
+    ```    
+
+The type of authentication as well as the credentials will be contained in the secret data specified by the Secret Name. Three types of authentication will be supported, which are `usernamepassword`, `apikey` and `bearertoken`. 
+
+- When `usernamepassword` is specified as the type in the secret data, the **basic-auth** `username:password@` will be inserted into the URI using the **username** and **password** found in the secret data.
+
+    !!! example - "Example secret data - `usernamepassword`"
+        ```
+        type=usernamepassword
+        username=myuser
+        password=mypassword
+        ```
+    !!! example  - "Example resulting URI with `basic-auth`"
+        ```
+        https://myuser:mypassword@example.com/uom.yaml
+        ```
+
+- When `apikey` is specified as the type in the secret data,  the **API Key** will be placed in the HTTP header using `apikeyname` and `apikeyvalue`  from the secret data
+
+    !!! example - "Example secret data - `apikey`"
+        ```
+        type=apikey
+        apikeyvalue=mykeyvalue
+        apikeyname=myname
+        ```

```

Comment:
How is API Key non-standard?
Nope, we didn't have it specifically specify it in the UCR, but that doesn't mean a design can't include it.

Reply:
Yea, the in query option should be included.

Reply:
> How is API Key non-standard? 

It is some custom format whereas Authorization: Basic and Bearer are standardized i.e. specified by IETF which is a standardisation body.

> Nope, we didn't have it specifically specify it in the UCR, but that doesn't mean a design can't include it.

That's actually one of the reasons we do have UCRs, to decide on what exactly is required for what use case before diving into design. Otherwise, we can add more and more features without anyone actually needing or using them.

Reply:
@farshidtz , FYI I went with @bnevis-i approach and eliminated the specific API-key option